src/main/java/com/cloudbees/jenkins/plugins/kubernetes_credentials_provider/KubernetesCredentialProvider.java

```diff
@@ -96,16 +110,24 @@ public void startWatchingForSecrets() {
 
             // XXX https://github.com/fabric8io/kubernetes-client/issues/1014
             // watch(resourceVersion, watcher) is deprecated but there is nothing to say why?
-            client = _client;
             LOG.log(Level.FINER, "registering watch");
-            watch = _client.secrets().withLabel(SecretUtils.JENKINS_IO_CREDENTIALS_TYPE_LABEL).watch(list.getMetadata().getResourceVersion(), this);
+            watch = _client.secrets().withLabel(SecretUtils.JENKINS_IO_CREDENTIALS_TYPE_LABEL).watch(this);
```

Comment:
this will introduce a regression as it will start the watch for any secrets created/updated from *now* but we retrieved the list of secrets earlier (thus if you create / update a secret in between line 98 and here they will be lost)

Reply:
you still have race conditions which is why you can pass the resource id.
has the API been removed in the latest version of the k8s plugin?